usr/lib/cca_stdll/cca_specific.c

```diff
@@ -2368,6 +2368,7 @@ CK_RV ccatok_hmac(STDLL_TokData_t * tokdata, SIGN_VERIFY_CONTEXT * ctx,
         TRACE_ERROR("%s\n", ock_err(ERR_OPERATION_NOT_INITIALIZED));
         return CKR_OPERATION_NOT_INITIALIZED;
     }
+    cca_ctx = (struct cca_sha_ctx *) ctx->context;
```

Comment:
this change does nothing, even more it splits one clear line of code into two.
What do you want to fix here ?

Reply:
The problem was that ctx->context was done in local variable declaration, before ctx is checked for being not NULL.  So in case ctx is NULL, then you have a null pointer dereference.  The new code avoids this by moving the dereferencing after the NULL checking.

Reply:
hm, let's assume ctx = NULL then the line
struct cca_sha_ctx *cca_ctx = (struct cca_sha_ctx *) ctx->context;
does introduce a new variable cca_ctx which is a pointer to ctx + some offset.
So if ctx is NULL, cca_ctx points to NULL + some offset.
As soon as you access something via this pointer you have a problem, but the pointer declaration and assignment is not a dereferencing by itself. In my opinion the old code here was and is correct and clearer to understand. The only point Coverty has in my mind is that when you try to check our derived pointer e.g.
if (!cca_ctx)  or  if (cca_ctx != NULL) or so, you will not get the right result.